pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/JClassSymbol.java

```diff
@@ -247,6 +250,22 @@ default Set<String> getAnnotationAttributeNames() {
                        .orElse(RetentionPolicy.CLASS);
     }
 
+    /**
+     * Return whether annotations of this annotation type apply to the
+     * given construct, as per the {@link Target} annotation. Return
+     * false if this is not an annotation.
+     */
+    default boolean annotationAppliesTo(ElementType elementType) {
+        if (!isAnnotation()) {
+            return false;
+        }
+        SymAnnot target = getDeclaredAnnotation(Target.class);
+        if (target == null) {
+            return false;
```

Comment:
This is not the right behavior… as per [the docs](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/Target.html):

> If an `@Target` meta-annotation is not present on an annotation type T , then an annotation of type T may be written as a modifier for any declaration except a type parameter declaration.

So, this should probably become:

```suggestion
        SymAnnot target = getDeclaredAnnotation(Target.class);
        if (target == null) {
            return elementType != ElementType.TYPE_PARAMETER;
```

Reply:
Thanks, I had no idea what the default was

Reply:
I should add unit tests for this as well